herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -1602,6 +1621,39 @@ public TableSpaceCheckpoint(
             this.tablesCheckpoints = tablesCheckpoints;
         }
     }
+    
+    public void createAndWriteTableDigest(TableSpaceManager manager, String tableSpace,String table ){
+        CommitLogResult pos;
+        boolean lockAcquired = false;   
+        StatementEvaluationContext context =  StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(); 
+
+        //Acquisisco il lock sul tablespace...devo bloccare le scritture
+        if (context.getTableSpaceLock() == 0) {
+            long lockStamp = acquireWriteLock("checkDataIntegrity");
+            context.setTableSpaceLock(lockStamp);
+            lockAcquired = true;
+        }      
+        try{
+            //long id = newTransactionId.getAndIncrement();
+            long id = 0;
+            long digest = TableDataChecksum.createChecksum(manager, tableSpace, table); 
+            if(digest != 0){
+                //write digest to LogEntry
+                LogEntry entry = LogEntryFactory.dataIntegrity(table,id,Bytes.from_long(digest));
```

Comment:
DigestData digestData = new DigestData();
digestData.setChecksum...
digestData.setChecksumType
digestData.setNumRecords

ObjectMapper mapper = new ObjectMapper();
byte[]  serialized = mapper.writeValueAsXXX(digestData);

Reply:
In LogEntry class i have a costructor that accept a bytes value, not a byte array ...
Is there a way to write an array of bytes in the logEntry?
   ```
 long digest = TableDataChecksum.createChecksum(manager, tableSpace, table); 
            if(digest != 0){
                DigestData digestData = new DigestData();
                digestData.setDigest(digest);
                digestData.setDigestType(TableDataChecksum.HASH_TYPE);
                digestData.SetNumRecords(numRecords);
                digestData.setTableName(table);
                digestData.setTableSpaceName(tableSpace);
                
                ObjectMapper mapper = new ObjectMapper();
                byte[] serialize = mapper.writeValueAsBytes(digestData);   
                
                //write digest to LogEntry
                LogEntry entry = LogEntryFactory.dataIntegrity(table,id, Bytes.from_long(digest));
                pos=log.log(entry, false);
                //apply with recory=false
                apply(pos, entry, false);
                
                return new DigestData(tableSpaceName, table, digest, digestType, numRecords, nextAutoIncrementValue);
                
            }

```